### PR TITLE
fix modifications

### PR DESCRIPTION
**Pull request by fgrunewald**
This fixes issue #384. However, there is one caveat: Now the modification can have an order attribute. This originates from the way we treat the atoms when making interactions. Even though we maybe don't want this, having order attributes is most consistent in my opinion with the fact that modifications is a subclass of `Link`. I would propose to later down the line disentangle the Modifications object from the Link, because they are really very separate things. As far as I can asses it makes no difference between having order attributes as to not having them in the current code.

### CODE REVIEW

File: vermouth/ffinput.py [hunk 1]
```diff
@@ -409,7 +409,7 @@ def _modification_atoms(self, line, lineno=0):
         tokens = collections.deque(_tokenize(line))
         _parse_link_atom(tokens, self.current_modification,
                          defaults={'PTM_atom': False},
-                         treat_prefix=False)
+                         treat_prefix=True)
```
Comment (pckroon) [inline vermouth/ffinput.py:412]: Why this change?

**Review by pckroon — DISMISSED** [review]
Small question, but lgtm otherwise. I agree that some restructuring is required, but that can be postponed for quite some time.

File: vermouth/ffinput.py (hunk 1 above)
Comment (fgrunewald) [reply vermouth/ffinput.py:412]: because otherwise you end up with some nodes having the order attribute and others not having the order attribute.

Comment (fgrunewald) [reply vermouth/ffinput.py:412]: but apparently it messes with some part of the code

Comment (pckroon) [reply vermouth/ffinput.py:412]: Ok. I guess that will get fixed at some point in the not-so-near future when we redo the parsers again.

Comment (fgrunewald) [reply vermouth/ffinput.py:412]: @pckroon can you see why this breaks the code? We either have the option to see why the order attribute on modifications messes with the code or we have to copy the entire base parser for interactions again only for modifications. The order attribute is so hard-coded into the interactions parser it is not even funny anymore. This is the last PR that has to be done for publication.
